backend/routes/giphy.js

```diff
@@ -11,8 +10,8 @@ router.route("/").get(async (req, res) => {
       src: img?.images?.downsized?.url || "",
       placeholderSrc: img?.images?.fixed_height_small_still?.url || ""
     }));
-
-    res.json(dataWithSrc);
+    const filterArr = Utils(dataWithSrc);
+    res.json(filterArr);
```

Comment:
I think it would be better to do filtering on frontend side. Because frontend is requesting data from backend with 'LIMIT'. According to this limit frontend creates Grid. So filtering elements on backend might/will send frontend misleading amount of images. 

What I would recommend is to use `Array.filter` on `Gallery.jsx` line 44.

What do you think?

Reply:
I think no. Because we eliminated duplicate on backend side. So duplicate data will never go to frontend.

Reply:
Okay will do changes on frontend.